File: datasetLoggi/JsonToFILOInstance.js
const fs = require("fs");

let capacity;
let points = [];
demand = [];

if (process.argv.length < 4) {
  console.log("Parametros incorretos\n");
  console.log("Primeiro parametro: arquivo a ser convertido\n");
  console.log("Destino do arquivo convertido\n");
  process.exit(1);
}

let json = fs.readFileSync(`${process.argv[2]}`);
let converted = JSON.parse(json);
capacity = converted.vehicle_capacity;
let originPoint = {
  lng: converted.origin.lng,
  lat: converted.origin.lat,
};
points.push(originPoint);
demand.push(0);

for (var i = 0; i < converted.deliveries.length; i++) {
  points.push(converted.deliveries[i].point);
  demand.push(converted.deliveries[i].size);
}

var logger = fs.createWriteStream(`${process.argv[3]}.txt`);
const R = 6371;

let originalFileName = process.argv[2].split("/");
originalFileName = originalFileName[originalFileName.length - 1].split(".");

logger.write(`NAME : ${originalFileName[0]} \n`);
logger.write(`COMMENT : \n`);
logger.write(`TYPE : 	CVRP \n`);
logger.write(`DIMENSION: ${converted.deliveries.length + 1} \n`);
logger.write(`EDGE_WEIGHT_TYPE : 	EUC_2D \n`);
logger.write(`CAPACITY : 	${capacity}\n`);
logger.write(`NODE_COORD_SECTION\n`);
for (var i = 0; i < points.length; i++) {
  logger.write(`${i + 1} ${points[i].lng} ${points[i].lat}\n`);
}
logger.write(`DEMAND_SECTION\n`);
for (var i = 0; i < demand.length; i++) {
  logger.write(`${i + 1}   ${demand[i]}\n`);
}
logger.write(`DEPOT_SECTION\n`);
logger.write(`${converted.origin.lng}\n`);
logger.write(`${converted.origin.lat}\n`);
logger.write(`EOF`);

File: datasetLoggi/JsonToFILOInstance.py
import json
import sys

if len(sys.argv) < 3:
    print("Parametros incorretos\n")
    print("Primeiro parametro: arquivo a ser convertido\n")
    print("Destino do arquivo convertido\n")
    sys.exit()

filePath = sys.argv[1]

# ABRE ARQUIVO LIDO DOS ARGUMENTOS
with open(filePath, "r") as data_file:
    json_data = data_file.read() 

data = json.loads(json_data)

# NOME DA INSTANCIA
nome = data["name"]

# CAPACIDADE DO VEICULO
capacidade = data["vehicle_capacity"]

# DADOS DAS ENTREGAS
deliveries = data["deliveries"]

# COORDENADAS DEPOSITO
originlng = data["origin"]["lng"]
originlat = data["origin"]["lat"]

# CABECARIO DO ARQUIVO DE SAIDA
output = open(sys.argv[2] + ".txt", "w")
output.write(f"NAME : {nome} \n")
output.write(f"COMMENT : \n")
output.write(f"TYPE : CVRP \n")
output.write(f"DIMENSION : {len(deliveries)+1}\n")
output.write(f"EDGE_WEIGHT_TYPE : EUC_2D \n")
output.write(f"CAPACITY : {capacidade} \n")
output.write(f"NODE_COORD_SECTION \n")
output.write(f"1 {originlng} {originlat}\n")

# COORDENADAS DAS ENTREGAS
counter = 2
for delivery in deliveries:
    output.write(f"{counter} {delivery['point']['lng']} {delivery['point']['lat']} \n")
    counter += 1

# SEÇÃO DE DEMANDA
output.write(f"DEMAND_SECTION\n")
output.write(f"1 0\n")
counter = 2
for delivery in deliveries:
    output.write(f"{counter} {delivery['size']} \n")
    counter += 1

# DEPOT SECTION E FINAL DO ARQUIVO
output.write(f"DEPOT_SECTION\n")
output.write(f"1\n-1\nEOF")
